batch/batch/disk.py

```diff
@@ -0,0 +1,120 @@
+import uuid
+import logging
+
+from hailtop.utils import sleep_and_backoff, check_shell
+
+log = logging.getLogger('disk')
+
+
+class Disk:
+    @staticmethod
+    async def delete_disk(compute_client, name, zone):
+        log.info(f'deleting disk {name}')
+        resp = None
+        delay = 0.2
+        delete_uuid = str(uuid.uuid4())
+        while resp is None or resp['status'] != 'DONE':  # pylint: disable=unsubscriptable-object
+            resp = await compute_client.delete(
+                f'/zones/{zone}/disks/{name}',
+                params={'requestId': delete_uuid})
+            delay = await sleep_and_backoff(delay)
+        log.info(f'deleted disk {name}')
+
+    def __init__(self, compute_client, name, zone, project, instance_name, size_in_gb, mount_path):
+        assert size_in_gb >= 10
+        assert len(name) <= 62
+
+        self.compute_client = compute_client
+        self.name = name
+        self.zone = zone
+        self.project = project
+        self.instance_name = instance_name
+        self.size_in_gb = size_in_gb
+        self.mount_path = mount_path
+
+        self.disk_path = f'/dev/disk/by-id/google-{self.name}'
+
+    async def __aenter__(self, labels=None):
+        await self.create(labels)
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.delete()
+
+    async def create(self, labels=None):
+        await self._create(labels)
+        await self._attach()
+        await self._format()
+
+    async def delete(self):
+        await self._detach()
+        await self._delete()
+
+    async def _format(self):
+        code = f'''
+mkfs.ext4 -m 0 -E lazy_itable_init=0,lazy_journal_init=0,discard {self.disk_path}
```

Comment:
For my own education, can you share what these flags mean and why you chose these settings?

Reply:
They're from the GCE disk tutorial on how to attach and format a new disk. I didn't actually look into if these parameters are optimal.